.github/workflows/main.yml

```diff
@@ -46,9 +47,17 @@ jobs:
           fi
       - name: Run Go test
         run: go test -v -timeout=30m ./...
-      - name: Run Go test -race
+                -cover -coverpkg=./... -covermode=count -coverprofile=gocoverage.txt
+      - name: Run Go test -race # note that -race can easily make the crypto stuff 10x slower
         if: github.ref == 'refs/heads/stage' ||  startsWith(github.ref, 'refs/heads/release')
-        run: go test -vet=off -timeout=30m -race ./... # note that -race can easily make the crypto stuff 10x slower
+        run: go test -vet=off -timeout=30m -race ./...
+                -cover -coverpkg=./... -covermode=atomic -coverprofile=gocoverage.txt
+
+      - name: Store code coverage artifact
+        uses: actions/upload-artifact@v4
```

Comment:
Shouldn't this be executed  under the same conditions that the `-race` tests are executed (stage/release) ?